app/views/organizations/show.html.erb

```diff
@@ -1,14 +1,27 @@
-<h1><%= @organization.name %></h1>
-<p><%= @organization.description %></p>
-<%= link_to @organization.address,
-  "https://www.google.com/maps/search/?api=1&query=#{@organization.address.to_param}",
-  target: '_blank'
-%>
+<div class="description"><%= @organization.description %></div>
 
-<ul>
-  <% @organization.needs.each do |need| %>
-    <li>
-      <%= need.item %>
-    </li>
+<div class="address">
+  <%= link_to @organization.address,
+    "https://www.google.com/maps/dir/?api=1&destination=#{@organization.address.to_param}",
+    target: '_blank'
+  %>
+</div>
+
+<h2 class="needs">Current Needs</h2>
+<div class="needs-list">
+  <% @needs.group_by(&:category).each do |category, needs| %>
+    <h3 class="category"><%= category.name %></h3>
+    <ul class="category">
+      <% needs.each do |need| %>
+        <li class="need">
+          <%= need.item %>
+          <% unless need.comment.nil? or need.comment =~ /^\s*$/ %>
```

Comment:
This could be `unless need.comment.blank?` instead, the behavior should be identical to your code and is more idiomatic.

Side note: `||` is generally preferred over `or` in Ruby, though they are identical except for precedence.

Reply:
That would definitely be better. Somehow the first answer I found when googling how to check for whitespace was to use a regex. The blank function is what I was looking for.